### PR TITLE
Fix --resume to skip to correct training step

### LINKED ISSUE #126: --resume doesn't work despite correctly loading from saved state
I'm trying to fix it so I can resume training from an existing WAN Lora checkpoint but it just doesn't seem to work. 
I checked other issues here but none of the proposed solutions worked:
- using --networks_weights instead of using --resume doesn't work for me
- proposed solution from kohya to use `--network_args "include_patterns=[r'.*(img_mod|txt_mod|modulation).*']"`  also didn't work

this is the command I run
`accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 wan_train_network.py --task t2v-14B --dit /workspace/ComfyUI/models/diffusion_models/wan2.1_t2v_14B_bf16.safetensors --dataset_config /workspace/musubi-tuner/wan/configs/config.toml --sdpa --mixed_precision bf16 --fp8_base --optimizer_type adamw8bit --learning_rate 3e-5 --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers --network_module networks.lora_wan --network_dim 16 --network_alpha 16 --timestep_sampling shift --discrete_flow_shift 3.0 --max_train_epochs 1000 --save_every_n_epochs 10 --save_state --t5 /workspace/diffusion-pipe/ckpt_path/wan2/models_t5_umt5-xxl-enc-bf16.pth --logging_dir /workspace/ComfyUI/models/loras/WAN2_LORAS/14B-musubi/lora_musubi_r16-16_lr3e5/tensorboard --output_dir /workspace/ComfyUI/models/loras/WAN2_LORAS/14B-musubi/lora_musubi_r16-16_lr3e5 --output_name lora_r16-16_1024x1024 --resume /workspace/ComfyUI/models/loras/WAN2_LORAS/14B-musubi/lora_musubi/lora_musubi-000010-state
`

most notable in this big ass command are:
`--save_state` this saves the state date to resume from
`--resume DIR` with path to the directory which has the state to resume from (it has the .bin file and some other stuff in there)

it seems to log that it correctly loaded the state from the path, but then proceeds to start from step 0.

```
INFO:hv_train_network:resume training from local state: /workspace/ComfyUI/models/loras/WAN2_LORAS/14B-musubi/lora_musubi_r16-16_lr3e5/lora_r16-16_1024x1024-000010-state
INFO:accelerate.accelerator:Loading states from /workspace/ComfyUI/models/loras/WAN2_LORAS/14B-musubi/lora_musubi_r16-16_lr3e5/lora_r16-16_1024x1024-000010-state
INFO:accelerate.checkpointing:All model weights loaded successfully
INFO:accelerate.checkpointing:All optimizer states loaded successfully
INFO:accelerate.checkpointing:All scheduler states loaded successfully
INFO:accelerate.checkpointing:All dataloader sampler states loaded successfully
INFO:accelerate.checkpointing:All random states loaded successfully
INFO:accelerate.accelerator:Loading in 0 custom states
```

### PR DESCRIPTION
**Pull request by DavidJBarnes**
Fix --resume to skip to correct training step

Fixes #126

 When resuming training with `--resume`, `global_step` and `epoch_to_start` were hardcoded to 0, causing training to re-run all steps from the beginning. This wastes GPU time and misaligns the LR schedule.

This reads the resumed step from `lr_scheduler.last_epoch` (which is populated by `accelerator.load_state()` from `scheduler.bin`) and:
  - Sets `global_step` and `epoch_to_start` correctly
  - Initializes the progress bar at the correct position
  - Skips already-trained batches when resuming mid-epoch

### CODE REVIEW

**Comment by marchcat69** [comment]
It works! Thank you.

**Comment by kohya-ss** [comment]
Thank you for this PR! Sorry for the slow review.

Your approach of getting the step count from the learning rate scheduler gave me a good starting point for thinking about this. However, after looking into it more closely, I found several issues that would need to be addressed:

- When gradient accumulation is enabled, `lr_scheduler.scheduler.last_epoch` appears to return the `global_step` value. `accelerator.step` (restored by `accelerator.load_state`) may be more appropriate as it tracks the data loader step.
- For datasets with a very large number of samples, skipping batches can take a significant amount of time, so this should probably be an opt-in option.
- The current implementation doesn't increment `global_step`, which could cause issues with training completion timing, sample image generation steps, etc.

Since these changes would essentially require a redesign, I'll close this PR and implement the feature myself in the near future, building on the ideas from your PR. Thank you again for tackling this, it's a real and important issue.
